### PR TITLE
#3477

### PR DESCRIPTION
**Pull request by vestrada**
https://github.com/craftercms/craftercms/issues/3477 - [studio-ui] Embedded Content DS should display Components when Content Type property is missing #3477

### CODE REVIEW

File: static-assets/components/cstudio-common/common-api.js [hunk 1]
```diff
@@ -259,7 +259,8 @@ var nodeOpen = false,
             HEADERS: "headers",
             AUTH_HEADERS: "AUTH_HEADERS",
             DATASOURCE_URL: "/static-assets/components/cstudio-forms/data-sources/",
-            CONTROL_URL: "/static-assets/components/cstudio-forms/controls/"
+            CONTROL_URL: "/static-assets/components/cstudio-forms/controls/",
+            GETALLCONTENTTYPES: "getAllContentType"
```
Comment (rart) [inline static-assets/components/cstudio-common/common-api.js]: Underscore the words

File: static-assets/components/cstudio-common/common-api.js [hunk 2]
```diff
@@ -2175,14 +2176,23 @@ var nodeOpen = false,
...
-            createNewContent: function(site, path, asPopup, formSaveCb, childForm, isFlattenedInclude) {
+            createNewContent: function(site, path, asPopup, formSaveCb, childForm, isFlattenedInclude, filter) {
                 var auxParams = [];
                 if(childForm && childForm == true) {
                     auxParams = [ { name: "childForm", value: "true" }];
                 }
 
                 var callback = {
                     success: function(contentTypes) {
+                        if (filter) {
```
Comment (rart) [inline static-assets/components/cstudio-common/common-api.js]: Make this a function that would be sent to array.prototype.filter

File: static-assets/components/cstudio-common/common-api.js (hunk 1 above)
Comment (rart) [reply static-assets/components/cstudio-common/common-api.js]: Remember to update all occurrences.

**Review by rart — CHANGES_REQUESTED** [review]
Please properly

File: static-assets/components/cstudio-common/common-api.js (hunk 1 above)
Comment (vestrada) [reply static-assets/components/cstudio-common/common-api.js]: Done

File: static-assets/components/cstudio-common/common-api.js (hunk 2 above)
Comment (vestrada) [reply static-assets/components/cstudio-common/common-api.js]: done

Comment (rart) [reply static-assets/components/cstudio-common/common-api.js]: This wasn’t the idea. The idea was that the consumer could send the filtering FUNCTION that you’d directly pass to array.filter. This way the filter is dynamic and not fixed to filtering the `.type` property.

File: static-assets/components/cstudio-common/common-api.js [hunk 3]
```diff
@@ -259,7 +259,8 @@ var nodeOpen = false,
             HEADERS: "headers",
             AUTH_HEADERS: "AUTH_HEADERS",
             DATASOURCE_URL: "/static-assets/components/cstudio-forms/data-sources/",
-            CONTROL_URL: "/static-assets/components/cstudio-forms/controls/"
+            CONTROL_URL: "/static-assets/components/cstudio-forms/controls/",
+            GET_ALL_CONTENTTYPES: "getAllContentType"
```
Comment (rart) [inline static-assets/components/cstudio-common/common-api.js]: Content types are two words. Still missing underscore.

**Review by rart — CHANGES_REQUESTED** [review]

**Review by rart — APPROVED** [review]

File: static-assets/components/cstudio-common/common-api.js (hunk 2 above)
Comment (vestrada) [reply static-assets/components/cstudio-common/common-api.js]: Done

File: static-assets/components/cstudio-common/common-api.js (hunk 3 above)
Comment (vestrada) [reply static-assets/components/cstudio-common/common-api.js]: Done

**Pull request merged**